### PR TITLE
Use internal slots

### PR DESCRIPTION
**Pull request by ptomato**
Closes: #231.

Open question from that issue: what about `dateTime.with(date)`, should that use the internal slots of `date`?

### CODE REVIEW

File: spec/abstractops.html [hunk 1]
```diff
@@ -82,8 +82,11 @@ <h1>ToDurationLike ( _duration_ )</h1>
       1. If Type(_duration_) is not Object, then
         1. Throw a *TypeError* exception.
       1. For each row of <emu-xref href="#table-temporal-durationlike-properties"></emu-xref>, except the header row, in table order, do
-        1. Let _property_ be the Property value of the current row.
-        1. Let _value_ be ? ToInteger(? Get(_duration_, _property_)).
+        1. If _duration_ has an [[InitializedTemporalDuration]] internal slot, then
+          1. Let _value_ be the value of _duration_'s internal slot whose name is the Internal Slot value of the current row.
```
Comment (gibson042) [inline spec/abstractops.html]: Are we **sure** we want conditional Get bypass on an argument? [`Array.prototype.concat`](https://tc39.es/ecma262/#sec-array.prototype.concat) could have done this but did not, and cases that do seem limited to signatures that _require_ typed input (e.g., [`new DataView`](https://tc39.es/ecma262/#sec-dataview-buffer-byteoffset-bytelength)). I'm not convinced that the benefits of such branches are sufficient to justify the complexity of including them.

Comment (ptomato) [reply spec/abstractops.html]: Indeed, I'm still wondering about code like what I wrote in [this comment](https://github.com/tc39/proposal-temporal/issues/231#issuecomment-577495048)...

**Comment by ptomato** [comment]
Since #362 landed, this is going to need a bit of revision.

**Comment by ptomato** [comment]
Revised. I ended up removing the changes to Duration since #362 already made the spec so that internal slots are consulted when passing in a Duration object.

The next step is still to determine whether we actually want this or not...

**Comment by ptomato** [comment]
Meeting, Feb. 27: We don't actually want this, at least not in this form.

**Comment by ljharb** [comment]
@ptomato can you elaborate? i very much want all important data for a temporal type to be handled with internal slots.

**Pull request closed without merging**